### DISCUSSION TITLE
Is it possible to use CrazyS for simulation to reality purpose?

### DISCUSSION THREAD

**Discussion by adededede** [gsilano/CrazyS#102] [Q&A]
I'm trying to developed some sort of swarm of crazyflies and thought of CrazyS for the simulation, however I was wondering if it was possible to use CrazyS simulation and port them to reality on real crazyflie?
I'm quiet new to these fields and I'm not sure of how to do that, or even if it's possible?
Thanks in advance for your answers

**Comment by gsilano** [comment]
Hey there! If you're looking to test your swarm algorithm, CrazyS is definitely worth checking out. This is the launch file with a swarm of drones: https://github.com/gsilano/CrazyS/blob/master/rotors_gazebo/launch/crazyflie2_swarm_hovering_example.launch
Just a heads up though, CrazyS hasn't been updated with the latest release of Crazyflie firmware. And to be honest, nobody has really tested how well the algorithm works in real-world scenarios.

But if you wanna fly in reality, I've got a few suggestions for you. These ROS packages and simulation environments are the most up-to-date ones out there. 

- CrazyChoir: https://github.com/OPT4SMART/crazychoir
- Crazyswarm: https://crazyswarm.readthedocs.io/en/latest/
- Crazyswarm2: https://github.com/IMRCLab/crazyswarm2
- GymBullet: https://github.com/utiasDSL/gym-pybullet-drones
- AerialGim: https://github.com/ntnu-arl/aerial_gym_simulator
- Bitcraze simulator: https://github.com/bitcraze/crazyflie-simulation
- Crazyflie with NMPC: https://github.com/bcbarbara/crazyflie_nmpc
- Crazyflie MRAC: https://github.com/fjctp/crazyflie_mrac
- Miscellaneous: https://github.com/wuwushrek/sim_cf, https://github.com/cipherlab-poly/crazyflie-public

Crazyflie firmware: https://www.bitcraze.io/development/external-projects/
